pexpect/pty_spawn.py

```diff
@@ -752,10 +752,14 @@ def interact(self, escape_character=chr(29),
         child process in interact mode is duplicated to the given log.
 
         You may pass in optional input and output filter functions. These
-        functions should take a string and return a string. The output_filter
-        will be passed all the output from the child process. The input_filter
-        will be passed all the keyboard input from the user. The input_filter
-        is run BEFORE the check for the escape_character.
+        functions should take bytes array and return bytes array too. Even
+        with ``encoding='utf-8'`` support, meth:`interact` will always pass
+        input_filter and output_filter decoded value. You may need to wrap your
```

Comment:
Again, these are not 'decoded values'.

- Decode: bytes -> str
- Encode: str -> bytes

Reply:
Sorry, i just skimed through your comment and went to switch (de/en)code.

So, can we say that this "_encoded_ value" get to passed to filters? And, in python3, filters do pass bytes, thus i had thought that "encoded str" is the best term. How should we express this in python2/3 compatible way?

Reply:
@takluyver

Reply:
I'd just say "will always pass input_filter and output_filter bytes".

`bytes` is meaningful on Python 2 and 3 - on Python 3 it's the name of the separate bytes type, and on Python 2 it's an alias for `str`. So I think it's fine to use that name.